### PR TITLE
fix: support phx 1.5 properly

### PR DESCRIPTION
**Pull request by novaugust**
Adds the actual trace parts of telemetry tracing.

Blocked on the same initial questions in my first pr, which is, what order do these things happen in if something goes wrong?  Made an issue in Phx for it https://github.com/phoenixframework/phoenix/issues/3927

### CODE REVIEW

**Comment by novaugust** [comment]
> I'm not sure how people generally use it, but I suspect that in many cases people are setting the service option in their tracer, and this would override that.

Is there a way for me to get the default service for a tracer?

**Comment by novaugust** [comment]
I added the service config to close out #11 ^.^

**Comment by novaugust** [comment]
I can also go back to letting peeps specify `:service` by itself, or I can update it to do a keyword merge. I initially just wanted to do `service`, but noticed that datadog wasn't immediately sure what type of service it was, so added the `type` param (though it seems DD figured it out after a few minutes of requests)

**Comment by zachdaniel** [comment]
I don't think there is a way to get the config out of a tracer, but a PR to spandex would make that possible. I think that supporting both of those options is fine, but that we shouldn't make the default (as in if they specify no options) also set the service.

**Comment by novaugust** [comment]
Ah got it. If they don't set service, then we shouldn't either. will fix.

**Comment by novaugust** [comment]
Alright @zachdaniel I think that's it. I have this up in production. WRT "when should we stop tracing?" I went the conservative route and call finish_trace in both router exception and endpoint stop. Also included documentation that phoenix doesn't actually publish all the events its documentation would make a user think it does, and so `Plug.Telemetry` is required for this to function correctly

File: lib/spandex_phoenix.ex [hunk 1]
```diff
@@ -235,17 +241,35 @@ defmodule SpandexPhoenix do
     :erlang.raise(kind, reason, stack)
   end
 
-  # Private Helpers
+  @doc false
+  def trace_request(conn, opts) do
```
Comment (novaugust) [inline lib/spandex_phoenix.ex:245]: errrrrrr it looks like i'm no longer using this in my code and should put it back where i grabbed it from

**Comment by novaugust** [comment]
Let me put the latest changes (revert starttrace changes) up in production and confirm we're still good

**Comment by novaugust** [comment]
Yup, looks good in prod. Ready for review. Sorry about the 🥚 🤕 situation last go-around

**Comment by zachdaniel** [comment]
Don't worry about it :) I haven't had time to work on spandex so its contributors like you that keep things going!

**Comment by zachdaniel** [comment]
🚀 Thank you for your contribution! 🚀

**Comment by zachdaniel** [comment]
Changes published as: `1.0.1`

**Pull request merged**